src/update_map.rs

```diff
@@ -72,6 +75,10 @@ impl<T: Clone> UpdateMap<T> for BTreeMap<usize, T> {
         Some(Cow::BTree(cow))
     }
 
+    fn get_arc(&self, k: usize) -> Option<Arc<T>> {
+        self.get(&k).cloned().map(Arc::new)
```

Comment:
This is unfortunate because it's just creating a new `Arc`.

Maybe we could have this as a fallback impl, and for maps that support it (like `BTreeMap<usize, Arc<T>>` we can have an efficient impl. The problem will be overlapping impls. We could get around that with a new type like `struct ArcMap<T>(VecMap<usize, Arc<T>>)`.

This is all turning out to be a bit complicated haha